### ISSUE TITLE
Permission denied when accessing image metadata

### ISSUE THREAD

**Issue by glyn** [projectriff/riff#1009]
I can reproduce this behaviour predictably whereas some others cannot. This may be because my roles (`Kubernetes Engine Admin`, `Owner`, `Storage Admin`) in the GCP project are different from those of others (who typically only have role `Owner`).

### Symptom

When I try to create a function, the build fails with the following error (where `$GCP_PROJECT` replaces the actual project name):
```
default/square-00001-vfvjh[build-step-analyze]: 2018/12/07 11:43:15 Error: failed to access image metadata from image : failed to access manifest gcr.io/$GCP_PROJECT/square: DENIED: "Permission denied for \"latest\" from request \"/v2/$GCP_PROJECT/square/manifests/latest\". "
```

### How to reproduce

Follow [Getting started on GKE](https://projectriff.io/docs/getting-started/gke/) and, when you come to the "GCR" section, ensure that you create a *new* GCP service account

Add the `--verbose` flag to the `riff function create` to see the above diagnostic more easily.

**Comment by glyn** [comment]
Could not reproduce using a fresh GCP project.

**Comment by glyn** [comment]
Re-opening since others have seen the same problem on GKE.

**Comment by mdcarlson** [comment]
Using the "[Getting Started on GKE](https://projectriff.io/docs/getting-started/gke/)" instructions to create a fresh cluster, I can consistently reproduce this behavior.   Happy to provide additional details on roles and config if that would be helpful

**Comment by glyn** [comment]
@mdcarlson Yes, please.

**Comment by mdcarlson** [comment]
@glyn, what would be helpful to have?  Project config and roles?

**Comment by glyn** [comment]
@mdcarlson Yes, I think project config and roles could give some valuable clues, thanks.

**Comment by mdcarlson** [comment]
I scripted the entire process from GCP project creation through invoking a function using Riff. Running this fully automated approach, I could not reproduce the "permission denied" error.

However, while developing this script, I did have some missteps in configuring the service account / role binding and keys that, even when cleaned up, seemed to cause the error.

Also, unsurprisingly, using the an existing, but incorrect, GCP_PROJECT_ID when creating and invoking the function will produce this error.

Now, that I CANT reproduce it with a fresh project/cluster/riff install, I'll try to figure out what went wrong in the existing project that I used initially.

**Comment by glyn** [comment]
Thanks for your efforts on this @mdcarlson. You may end up where we did: no idea what the salient difference is between the clusters that work and those that don't.

**Comment by mdcarlson** [comment]
I nuked and recreated the service account, clusters and roles in my original project and experienced the permissions denied issue again.  I used the same code from my scripted approach for consistency.

That points to a configuration issue of some type with my old project that existed before I ran the steps to create the other artifacts.  Since I didn't really use that project for much, it is not worth my troubleshooting further, but if I stumble across an answer, I'll be sure to share.

In summary, the steps in the "Getting Started on GKE" worked every time when applied to a new project, but when applied to my existing project, I ran into this "permission denied" issue when attempting to create the sample square function using "riff create..."
